index.js

```diff
@@ -140,6 +125,52 @@ class Action {
 
         this._checkForUpdate()
     }
+
+    run() {
+        if (!this.solutionFile || !fs.existsSync(this.solutionFile)) {
+            this._printErrorAndExit("solution file not found")
+        }
+
+        if (!this.packagePath || !fs.existsSync(this.packagePath)) {
+            this._printErrorAndExit("PACKAGE_PATH not provided.")
+        }
+
+        // nuke ane normal .nupkg/.snupkg inside the user specified package path
+        // and then build the packages for the resulting projects inside of the
+        // solution file specified.
+        fs.readdirSync(this.packagePath).filter(fn => /\.s?nupkg$/.test(fn)).forEach(fn => fs.unlinkSync(fn))
+        this._executeInProcess(`dotnet build -c Release ${this.solutionFile}`)
+        this._executeInProcess(`dotnet pack ${this.includeSymbols ? "--include-symbols -p:SymbolPackageFormat=snupkg" : ""} --no-build --no-restore -c Release ${this.solutionFile} -o ${this.packagePath}`)
+        const packages = fs.readdirSync(this.packagePath).filter(fn => fn.endsWith("nupkg"))
+        console.log(`Generated Package(s): ${packages.join(", ")}`)
+
+        // TODO: output all of the package names and paths.
+        const packageFilename = packages.filter(p => p.endsWith(".nupkg"))[0],
+            symbolsFilename = packages.filter(p => p.endsWith(".snupkg"))[0]
+        process.stdout.write(`::set-output name=PACKAGE_NAME::${packageFilename}` + os.EOL)
+        process.stdout.write(`::set-output name=PACKAGE_PATH::${path.resolve(packageFilename)}` + os.EOL)
+        if (symbolsFilename) {
+            process.stdout.write(`::set-output name=SYMBOLS_PACKAGE_NAME::${symbolsFilename}` + os.EOL)
+            process.stdout.write(`::set-output name=SYMBOLS_PACKAGE_PATH::${path.resolve(symbolsFilename)}` + os.EOL)
+        }
+
+        if (!hasGlob(this.projectFile) && !hasGlob(this.versionFile)) {
+            this._run_internal()
+        }
+
+        // it has a glob, now we need to recursively obtain all files
+        // represented in the glob and match them up. After that we
+        // need to reset the projectFile, and versionFile variables on
+        // the object instance each time and call _run_internal() for
+        // each file found that matches in the glob.
+        let tmp = this.projectFile
+        glob.on('include', function (file) {
+            this.projectFile = file.relative
+            this.versionFile = file.relative
```

Comment:
I was with you on every thing in the PR but this. Rather than mutating the current instance of the Action, make this a pure method, passing project and version into `_run_internal` as parameters.

Also, what if the version file is not the project file? What if the version is indicated in a .nuspec file? That was the intent of the versionfile property.

Reply:
The issue with file globbing is well the following:

- user uses file globbing for project file path (``'**/*.csproj'``)
- user uses file globbing for nuspec files (usually using nuspec files manually is not wanted by developers as they tend to go outdated fast and so they trick the .NET SDK to generate them instead).

For the nuspec case I think one could glob for them but then one would have to:
- filter the ``include`` event between ``csproj`` and ``nuspec`` files.

I guess there is an easier way to process the events and process the include event if I check if ``versionFile`` specifies an glob pattern and then resolve those first, then resolve the project globs and then pass them in properly.

But this results in other issues:
- what if some projects do not use nuspec, but some do on the same repository all of which package up packages to be published
- what would prevent a bug where the file globbing to fix the nuspec issue ends up resulting in it not wanting to read the ``version`` that is embedded in the projects (csprojs) that does not have a nuspec file paired with it?